pkg/scheduler/core/generic_scheduler.go

```diff
@@ -595,7 +595,9 @@ func addNominatedPods(pod *v1.Pod, meta predicates.PredicateMetadata,
 		if util.GetPodPriority(p) >= util.GetPodPriority(pod) && p.UID != pod.UID {
 			nodeInfoOut.AddPod(p)
 			if metaOut != nil {
-				metaOut.AddPod(p, nodeInfoOut)
+				if err := metaOut.AddPod(p, nodeInfoOut); err != nil {
```

Comment:
You should take this out of `nodeInfoOut` upon error, and update `podsAdded` to false.